apps/cli/test/api.test.ts

```diff
@@ -0,0 +1,647 @@
+          timeout: 10000,
+        },
+      );
+    });
+  });
+
+  describe("getCodemod", () => {
+    const mockName = "test-codemod";
+    const mockResponse: GetCodemodResponse = {
+      name: mockName,
+      arguments: [],
+      amountOfUses: 0,
+      applicability: null,
+      author: "test-author",
+      createdAt: new Date(),
+      engine: null,
+      featured: false,
+      frameworks: [],
+      frameworkVersion: null,
+      id: 3003,
+      hidden: false,
+      labels: [],
+      openedPrs: 0,
+      private: false,
+      shortDescription: "Test codemod",
+      slug: "test-codemod",
+      tags: [],
+      updatedAt: new Date(),
+      totalRuns: 0,
+      verified: false,
+      totalTimeSaved: 0,
+      useCaseCategory: null,
+      versions: [],
+    };
+
+    it("should get codemod with access token", async () => {
+      mockedAxios.get.mockResolvedValueOnce({ data: mockResponse });
+
+      const result = await getCodemod(mockName, {
+        accessToken: mockAccessToken,
+      });
+
+      expect(mockedAxios.get).toHaveBeenCalledWith(
+        `http://api.example.com/codemods/${mockName}`,
+        {
+          headers: { Authorization: `Bearer ${mockAccessToken}` },
+          timeout: 10000,
+        },
+      );
+      expect(result).toEqual(mockResponse);
+    });
+
+    it("should get codemod with API key", async () => {
+      mockedAxios.get.mockResolvedValueOnce({ data: mockResponse });
+
+      const result = await getCodemod(mockName, { apiKey: mockApiKey });
+
+      expect(mockedAxios.get).toHaveBeenCalledWith(
+        `http://api.example.com/codemods/${mockName}`,
+        {
+          headers: { "X-API-Key": mockApiKey },
+          timeout: 10000,
+        },
+      );
+      expect(result).toEqual(mockResponse);
+    });
+  });
+
+  describe("getGithubAPIKey", () => {
+    it("should get GitHub API key", async () => {
+      const mockToken = "github-token";
+      mockedAxios.get.mockResolvedValueOnce({ data: { token: mockToken } });
+
+      const result = await getGithubAPIKey(mockAccessToken);
+
+      expect(mockedAxios.get).toHaveBeenCalledWith(
+        "http://auth.example.com/oAuthToken",
+        {
+          headers: { Authorization: `Bearer ${mockAccessToken}` },
+          timeout: 10000,
+        },
+      );
+      expect(result).toBe(mockToken);
+    });
+  });
+
+  describe("getGithubAvailableScope", () => {
+    it("should get GitHub available scopes", async () => {
+      const mockScopes = ["repo", "user"];
+      const mockResponse = {
+        status: 200,
+        headers: {
+          "x-oauth-scopes": mockScopes.join(", "),
+        },
+      };
+
+      const mockRequestMethod = vi.fn().mockResolvedValueOnce(mockResponse);
+      mockedOctokit.mockImplementation(
+        () =>
+          ({
+            request: mockRequestMethod,
+            // biome-ignore lint/suspicious/noExplicitAny: This is a mock implementation
+          }) as any,
```

Comment:
I don't like this. But a concise way to bring in the rest of the Octokit implementation eluded me today